{{ cookiecutter.package_name }}/setup.py

```diff
@@ -1,33 +1,87 @@
+# other information are set in the setup.cfg file.
 
-# Ensure that astropy-helpers is available
-import ah_bootstrap  # noqa
+import os
+import sys
 
 from setuptools import setup
 from setuptools.config import read_configuration
+{% if cookiecutter.include_example_code == 'y' %}
+from extension_helpers import get_extensions
+{% endif %}
+
+# First provide helpful messages if contributors try and run legacy commands
+# for tests or docs.
+
+TEST_HELP = """
+Note: running tests is no longer done using 'python setup.py test'. Instead
+you will need to run:
+
+    tox -e test
+
+If you don't already have tox installed, you can install it with:
+
+    pip install tox
+
+If you only want to run part of the test suite, you can also use pytest
+directly with::
+
+    pip install -e .[test]
+    pytest
+
+For more information, see:
+
+  http://docs.astropy.org/en/latest/development/testguide.html#running-tests
+"""
+
+if 'test' in sys.argv:
+    print(TEST_HELP)
+    sys.exit(1)
+
+DOCS_HELP = """
+Note: building the documentation is no longer done using
+'python setup.py build_docs'. Instead you will need to run:
+
+    tox -e build_docs
+
+If you don't already have tox installed, you can install it with:
+
+    pip install tox
+
+You can also build the documentation with Sphinx directly using::
 
-from astropy_helpers.setup_helpers import register_commands, get_package_info
-from astropy_helpers.version_helpers import generate_version_py
+    pip install -e .[docs]
+    cd docs
+    make html
 
-# Store the package name in a built-in variable so it's easy
-# to get from other parts of the setup infrastructure
-builtins._ASTROPY_PACKAGE_NAME_ = read_configuration('setup.cfg')['metadata']['name']
+For more information, see:
 
-# Create a dictionary with setup command overrides. Note that this gets
-# information about the package (name and version) from the setup.cfg file.
-cmdclass = register_commands()
+  http://docs.astropy.org/en/latest/install.html#builddocs
+"""
 
-# Freeze build information in version.py. Note that this gets information
-# about the package (name and version) from the setup.cfg file.
-version = generate_version_py()
+if 'build_docs' in sys.argv or 'build_sphinx' in sys.argv:
+    print(DOCS_HELP)
+    sys.exit(1)
 
-# Get configuration information from all of the various subpackages.
-# See the docstring for setup_helpers.update_package_files for more
-# details.
-package_info = get_package_info()
+VERSION_TEMPLATE = """
+# Note that we need to fall back to the hard-coded version if either
+# setuptools_scm can't be imported or setuptools_scm can't determine the
+# version, so we catch the generic 'Exception'.
+try:
+    from setuptools_scm import get_version
+    version = get_version(root='..', relative_to=__file__)
+except Exception:
+    version = '{version}'
+""".lstrip()
 
-setup(version=version, cmdclass=cmdclass, **package_info)
+{% if cookiecutter.use_compiled_extensions == 'y' %}
```

Comment:
I have a less repetitive version of this in the OA guide.

Reply:
Fixed

Reply:
The generated version looks like:

```
setup(
    use_scm_version={'write_to': os.path.join('test_package', 'version.py'),
                     'write_to_template': VERSION_TEMPLATE},

    ext_modules=get_extensions()

)
```

though... Any ideas on avoiding that?

Reply:
I think changing `{%` to `{%-` should remove the extra whitespace or something like that.

Reply:
Ok that worked - thanks!